### DISCUSSION TITLE
How to order by a model with m-n relationship

### DISCUSSION THREAD

**Discussion by tomaszferens** [prisma/prisma#12509] [Q&A]
I have a following schema:

```prisma
model User {
  id          Int           @id @default(autoincrement())
  name        String

  userProjects UserProject[]
}

model Project {
  id          Int           @id @default(autoincrement())
  name        String

  userProjects UserProject[]
}

model UserProject {
  userId        Int
  user          User     @relation(fields: [userId], references: [id])
  projectId     Int
  project       Project  @relation(fields: [projectId], references: [id])
  lastVisitedAt DateTime

  @@id([userId, projectId])
  @@unique([userId, projectId])
}
```

The purpose of UserProject table is to store `lastVisitedAt`, so each user visited each project at a different time. They are unique.

Now, I have an issue with getting projects ordered by most recent date - `lastVisitedAt`:

```
prisma.project.findMany({
  where: {
    userId: user.id,
  },
  orderBy: {
    userProjects: ?? // How to write a query here?
  },
});
```

**Comment by austincrim** [comment]
Hey @tomaszferens 👋,

You could try querying through the `UserProject` model instead. 
Would something like this work?
```ts
await prisma.userProject.findMany({
  where: {
    userId: 123
  },
  select: {
    project: true
  },
  orderBy: {
    lastVisitedAt: 'desc'
  }
})
```

**Reply by vladi-stevanovic** [reply]
👋 Hello @tomaszferens  I'm going through pending GitHub Discussions to check if you need any help. Is this issue still open?
Please let us know if we can investigate it further or if you've already solved it, we'd love to know how!

It would be super useful for other users facing this problem! 🙌

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
